src/dom-utils.ts

```diff
@@ -88,6 +110,14 @@ export function isFocusable(el: HTMLElement) {
   )
 }
 
+function canHaveFocusableChildren(el: HTMLElement) {
+  return !(
+    el.matches(
+      '[hidden],[inert],:disabled,[aria-hidden="true"],[aria-disabled="true"]'
+    ) && el.matches(PRESENTATIONAL_CHILDREN_SELECTOR)
```

Comment:
Maybe we could adjust our presentational selector array?
```js
const PRESENTATIONAL_CHILDREN_SELECTOR = [
  'a[href]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  'button:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  'img:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  'summary:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="button"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="image"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="link"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="math"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="presentation"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="progressbar"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="scrollbar"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
  '[role="slider"]:not([inert]):not([hidden]):not(:disabled):not([aria-hidden="true"]):not([aria-disabled="true"])',
].join(',')
```

But I guess it’s not better. 😹

Reply:
I don't think we need this much detail to decide if an element can or cannot have presentational children, tbh. If I'm missing something, please let me know!